morph/cli.py

```diff
@@ -78,6 +80,23 @@ def db_path(db_name):
     return os.path.join(profile_path(), 'dbs', db_name + '.db')
 
 
+def load_db(db_name, allow_missing=False):
+    path = db_path(db_name)
+    if not allow_missing and not os.access(path, os.R_OK):
+        die('can\'t read db file: %s' % (path,))
+    return MorphDb(path, ignoreErrors=allow_missing)
+
+
+def parse_morpheme(morpheme_str):
+    fields = morpheme_str.split('\t')
+    if len(fields) != 4:
+        die('bad morpheme (should be 4 tab-separated fields, MorphMan-style): %s'
+            % (morpheme_str,))
+    # Only 4 fields show up in equality, hash, or show, but the constructor
+    # takes an extra one.  That's the second argument; duplicate the first.
```

Comment:
Congratulations, you found a serious design issue in MorphMan :)

The Morpheme class both contains the meaning as a "morpheme"* because it has an `inflected` attribute and the meaning as a "lexeme"*, because it does not include this attribute in comparisons of morphemes.

This combined with the memoization of `fixReading` in morphemizer.py sometimes memoizes words which are different morphemes but have the same lexemes. For example で and だ both get the same base form from だ (or で I don't remember) from mecab. Because of that, the memoization kicks in and returns the result of the で/だ morpheme that was first processed with this function.
This is visible in places where the morpheme interpretation is used, mainly highlighting (inflected forms get wrapped into a colored `<span>`, but で and だ are often colorless) and the `View Morphemes` menu item. But the wrong morpheme already gets returned from the getMorphemesFromExpr!
The rest of the codebase uses the lexeme interpretation (database code, note rating code).

The cleanest way is to create a `Lexeme` class, which does not have the `inflected` attribute and remove the __eq__ and __hash__ functions from the `Morpheme`.

This probably means that this project should have been called `LexMan` instead of `MorphMan` :)

\* morpheme: an inflected form of a word; so `go`, `went` and `going` are different morphemes
\* lexeme: the class of words that can be created through inflection; `go`, `went` and `going` have the same lexeme, `drink` and `go` are different lexemes